awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,238 @@
+import os.path
+
+from django.db import connection
+from django.db.models import Count
+from django.conf import settings
+from django.utils.timezone import now
+
+from awx.conf.license import get_license
+from awx.main.utils import (get_awx_version, get_ansible_version,
+                            get_custom_venv_choices, camelcase_to_underscore)
+from awx.main import models
+from django.contrib.sessions.models import Session
+from awx.main.analytics import register
+
+'''
+This module is used to define metrics collected by awx.main.analytics.gather()
+Each function is decorated with a key name, and should return a data
+structure that can be serialized to JSON
+
+@register('something')
+def something(since):
+    # the generated archive will contain a `something.json` w/ this JSON
+    return {'some': 'json'}
+
+All functions - when called - will be passed a datetime.datetime object,
+`since`, which represents the last time analytics were gathered (some metrics
+functions - like those that return metadata about playbook runs, may return
+data _since_ the last report date - i.e., new data in the last 24 hours)
+'''
+
+
+@register('config')
+def config(since):
+    license_info = get_license(show_key=False)
+    return {
+        'system_uuid': settings.SYSTEM_UUID,
+        'tower_url_base': settings.TOWER_URL_BASE,
+        'tower_version': get_awx_version(),
+        'ansible_version': get_ansible_version(),
+        'license_type': license_info.get('license_type', 'UNLICENSED'),
+        'free_instances': license_info.get('free instances', 0),
+        'license_expiry': license_info.get('time_remaining', 0),
+        'pendo_tracking': settings.PENDO_TRACKING_STATE,
+        'authentication_backends': settings.AUTHENTICATION_BACKENDS,
+        'logging_aggregators': settings.LOG_AGGREGATOR_LOGGERS
```

Comment:
This is interesting data:`settings.LOG_AGGREGATOR_LOGGERS`:

```
In [2]: settings.LOG_AGGREGATOR_LOGGERS
Out[2]: ['awx', 'activity_stream', 'job_events', 'system_tracking']
```

But we probably also want to know if logging is **enabled**, and if so, what type they're using:

```python
'external_logger_enabled': settings.LOG_AGGREGATOR_ENABLED,
'external_logger_type': getattr(settings, 'LOG_AGGREGATOR_TYPE', None),
```

Reply:
yeah, list is totally irrelevant if it's not enabled. That's only for sub-selection in case it is enabled.

Reply:
Ok, I figure Insights will check if it is enabled first, and if not, then they can just disregard the `LOG_AGGREGATOR_LOGGERS` setting.  

I've added this, here is an example of how it will show with loggly set up:

```
Out[103]: 
{'ansible_version': '2.7.8',
 'authentication_backends': ['awx.sso.backends.TACACSPlusBackend',
  'django.contrib.auth.backends.ModelBackend'],
 'external_logger_enabled': True,
 'external_logger_type': 'loggly',
 'free_instances': 0,
 'license_expiry': 8519115645,
 'license_type': 'enterprise',
 'logging_aggregators': ['awx', 'activity_stream', 'job_events'],
 'pendo_tracking': 'off',
 'system_uuid': '00000000-0000-0000-0000-000000000000',
 'tower_url_base': 'https://localhost:8043',
 'tower_version': '3.0.1-282-gd2dcd1186'}
```